python/cudf/cudf/core/column/string.py

```diff
@@ -2026,34 +2037,62 @@ def split(self, pat=None, n=-1, expand=None, **kwargs):
         >>> s
         0            this is a regular sentence
         1    https://docs.python.org/index.html
-        2                                  None
+        2                                  <NA>
         dtype: object
 
+        In the default setting, the string is split by whitespace.
+
+        >>> s.str.split()
+        0        [this, is, a, regular, sentence]
+        1    [https://docs.python.org/index.html]
+        2                                    None
+        dtype: list
+
+        Without the ``n`` parameter, the outputs of ``rsplit``
+        and ``split`` are identical.
+
+        >>> s.str.rsplit()
+        0        [this, is, a, regular, sentence]
+        1    [https://docs.python.org/index.html]
+        2                                    None
+        dtype: list
+
         The `n` parameter can be used to limit the number of
         splits on the delimiter.
 
         >>> s.str.split(n=2)
-                                            0     1                   2
-        0                                this    is  a regular sentence
-        1  https://docs.python.org/index.html  None                None
-        2                                None  None                None
+        0          [this, is, a regular sentence]
+        1    [https://docs.python.org/index.html]
+        2                                    None
+        dtype: list
 
         The `pat` parameter can be used to split by other characters.
 
-        >>> s.str.split(pat = "/")
-                                    0     1                2           3
-        0  this is a regular sentence  None             None        None
-        1                      https:        docs.python.org  index.html
-        2                        None  None             None        None
+        >>> s.str.split(pat="/")
+        0               [this is a regular sentence]
+        1    [https:, , docs.python.org, index.html]
+        2                                       None
+        dtype: list
+
+        When using ``expand=True``, the split elements will expand out
+        into separate columns. If ``<NA>`` value is present, it is propagated
+        throughout the columns during the split.
+
+        >>> s.str.split(expand=True)
+                                            0     1     2        3         4
+        0                                this    is     a  regular  sentence
+        1  https://docs.python.org/index.html  <NA>  <NA>     <NA>      <NA>
+        2                                <NA>  <NA>  <NA>     <NA>      <NA>
         """
-        if expand is None:
-            expand = True
-            warnings.warn("`expand` parameter defatults to True.")
-        elif expand is not True:
-            raise NotImplementedError(
-                "`expand=False` setting is not supported yet"
+        if expand not in (None, True, False):
```

Comment:
Any reason for expand defaulting to `None`, rather than `True` or `False`?

Reply:
This purely for pandas compatibility since pandas takes this param through `kwargs`

Reply:
I don't understand - how does defaulting to `None` help with Pandas compatibility?

Reply:
The reason I've added `None` as the default parameter is both `None`/`False` have the same behavior in pandas.

So right now I've defaulted to `None` but we can change it to `False` and also nicely handle even when `expand` is `None`.

Reply:
Yeah - would suggest we defaulted to `False`. That way, users can get that information from the signature line rather than have to look in the docstring.

Reply:
Fixed this